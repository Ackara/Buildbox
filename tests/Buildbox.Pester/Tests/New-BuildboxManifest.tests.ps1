Import-Module "$(Split-Path $PSScriptRoot -Parent)\config.psm1" -Force
$context = New-TestEnvironment;
Import-Module $context.ModulePath -Force;

Describe "New-BuildboxManifest" {
	Push-Location $context.TestDir;
	Context "Can Create File" {
		It "should create a new 'manifest.json' when no args are passed." {
			$manifest = New-BuildboxManifest;
			$manifest.Path | Should Exist;
		}

		It "should thow an exception when a existing file is being overwritten." {
			{ New-BuildboxManifest } | Should Throw;
		}

		It "should overwrite an existing 'manifest.json' file." {
			$manifest = New-BuildboxManifest -Force;
			$manifest.Path | Should Exist;
		}
	}

	Context "Has Required Fields" {
		$manifest = New-BuildboxManifest "$PWD\new-manifest1.json" -Force;
		
		It "should have all expected fields when serialized" {
			{ Approve-File($manifest.Path) } | Should Not Throw;
		}

		It "should be of type Manifest" {
			$manifest.GetType() | Should Be Manifest;
		}
	}
	Pop-Location;
}