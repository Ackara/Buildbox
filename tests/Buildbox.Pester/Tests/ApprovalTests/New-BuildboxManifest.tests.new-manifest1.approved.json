{
    "Id":  "CompanyName.ProductName",
    "Version":  {
                    "Major":  0,
                    "Minor":  0,
                    "Patch":  1
                },
    "ProductName":  "Your Product Name",
    "Description":  "A useful description.",
    "Authors":  "Ackeem",
    "Tags":  "tag1 tag2",
    "Owner":  "Ackeem",
    "Copyright":  "All Rights Reserved.",
    "IconUri":  null,
    "LicenseUri":  null,
    "ProjectUrl":  null,
    "RepositoryUrl":  null,
    "ReleaseNotes":  null,
    "BranchSuffixMap":  {
                            "*":  "beta",
                            "master":  ""
                        }
}
